frame/evm-system/src/lib.rs

```diff
@@ -165,28 +176,47 @@ impl<T: Config> Pallet<T> {
 
 	/// Create an account.
 	pub fn create_account(who: &<T as Config>::AccountId) -> AccountCreationOutcome {
-		if Self::account_exists(who) {
-			return AccountCreationOutcome::AlreadyExists;
-		}
-
-		Account::<T>::insert(who.clone(), AccountInfo::<_, _>::default());
-		Self::on_created_account(who.clone());
-		AccountCreationOutcome::Created
+		Account::<T>::mutate_exists(who, |maybe_account| match maybe_account {
+			Some(ref mut account) => {
+				account.sufficients = account.sufficients.saturating_add(1);
+				AccountCreationOutcome::AlreadyExists
+			}
+			new_account @ None => {
+				let mut account: AccountInfo<_, _> = Default::default();
+				account.sufficients = 1;
+				*new_account = Some(account);
+				Self::on_created_account(who.clone());
+				AccountCreationOutcome::Created
+			}
+		})
 	}
 
 	/// Remove an account.
 	pub fn remove_account(who: &<T as Config>::AccountId) -> AccountRemovalOutcome {
-		if !Self::account_exists(who) {
-			return AccountRemovalOutcome::DidNotExist;
-		}
-
-		if Account::<T>::get(who).data != <T as Config>::AccountData::default() {
-			return AccountRemovalOutcome::Retained;
-		}
-
-		Account::<T>::remove(who);
-		Self::on_killed_account(who.clone());
-		AccountRemovalOutcome::Reaped
+		Account::<T>::mutate_exists(who, |maybe_account| {
```

Comment:
There seems to be some duplication between this new implementation and the `try_mutate_exists` fn; can't they somehow be combined?